python/cuspatial/cuspatial/_lib/nearest_points.pyx

```diff
@@ -0,0 +1,78 @@
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+from cudf._lib.column cimport Column
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.column.column_view cimport column_view
+
+from cuspatial._lib.cpp.optional cimport nullopt, optional
+from cuspatial._lib.cpp.point_linestring_nearest_points cimport (
+    pairwise_point_linestring_nearest_points as c_func,
+    point_linestring_nearest_points_result,
+)
+
+
+def pairwise_point_linestring_nearest_points(
+    Column points_xy,
+    Column linestring_part_offsets,
+    Column linestring_points_xy,
+    multipoint_geometry_offset=None,
+    multilinestring_geometry_offset=None,
+):
+    cdef Column multipoint_geometry_offset_column
+    cdef Column multilinestring_geometry_offset_column
+    cdef optional[column_view] c_multipoint_geometry_offset
+    cdef optional[column_view] c_multilinestring_geometry_offset
+
+    if multipoint_geometry_offset is not None:
+        multipoint_geometry_offset_column = multipoint_geometry_offset
+        c_multipoint_geometry_offset = multipoint_geometry_offset_column.view()
```

Comment:
```suggestion
        c_multipoint_geometry_offset = multipoint_geometry_offset.view()
```

Reply:
Unfortunately, this will not work because `multipoint_geometry_offset` is not statically typed. In cython `multipoint_geometry_offset` is typed `PyObject *`. The type of `multipoint_geometry_offset.view()`, is thus also `PyObject *`. Cython compiler will complain assigning a `PyObject *` type to `optional[column_view]`.

Reply:
Let that be said, this whole pattern can be generalized into a cdef function. Will work on that.

Reply:
Done.